Upgrade Task failing

### Problem Statement

I have a live db that is failing to upgrade

```
16:11:08.124  INFO  runonce.Task230119MigrateContentToProperPersonaTagAndRemoveDupTags - ======================================================================
16:11:08.124  INFO  runonce.Task230119MigrateContentToProperPersonaTagAndRemoveDupTags - ======================================================================
16:11:08.124  INFO  runonce.Task230119MigrateContentToProperPersonaTagAndRemoveDupTags - ==> Executing upgrade script
16:11:08.124  INFO  runonce.Task230119MigrateContentToProperPersonaTagAndRemoveDupTags - ======================================================================
16:11:08.141  FATAL runonce.Task230119MigrateContentToProperPersonaTagAndRemoveDupTags - Unable to execute SQL upgrade
org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "tag_inode_pkey"
  Detail: Key (tag_id, inode)=(ab344fa0-c909-4b3c-aa8c-6bacd4f770fa, 7dbb1cb6-1058-46ff-adf4-21bce8f63216) already exists.
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2676) ~[postgresql-42.5.1.jar:42.5.1]
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2366) ~[postgresql-42.5.1.jar:42.5.1]
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:356) ~[postgresql-42.5.1.jar:42.5.1]
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:496) ~[postgresql-42.5.1.jar:42.5.1]
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:413) ~[postgresql-42.5.1.jar:42.5.1]
	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:333) ~[postgresql-42.5.1.jar:42.5.1]
	at org.postgresql.jdbc.PgStatement.executeCachedSql(PgStatement.java:319) ~[postgresql-42.5.1.jar:42.5.1]
	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:295) ~[postgresql-42.5.1.jar:42.5.1]
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:290) ~[postgresql-42.5.1.jar:42.5.1]
	at com.zaxxer.hikari.pool.ProxyStatement.execute(ProxyStatement.java:95) ~[HikariCP-3.4.2.jar:?]
	at com.zaxxer.hikari.pool.HikariProxyStatement.execute(HikariProxyStatement.java) ~[HikariCP-3.4.2.jar:?]
	at com.dotmarketing.common.db.DotConnect.executeStatement(DotConnect.java:264) ~[classes/:?]
	at com.dotmarketing.startup.AbstractJDBCStartupTask.executeUpgrade(AbstractJDBCStartupTask.java:790) ~[classes/:?]
	at com.dotmarketing.startup.StartupTasksExecutor.executeUpgrades(StartupTasksExecutor.java:199) ~[classes/:?]
	at com.liferay.portal.servlet.MainServlet.init(MainServlet.java:119) ~[classes/:?]
	at org.apache.catalina.core.StandardWrapper.initServlet(StandardWrapper.java:1164) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.StandardWrapper.loadServlet(StandardWrapper.java:1117) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.StandardWrapper.load(StandardWrapper.java:1010) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.StandardContext.loadOnStartup(StandardContext.java:4957) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5264) ~[catalina.jar:9.0.60]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:726) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:698) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:696) ~[catalina.jar:9.0.60]
	at org.apache.catalina.startup.HostConfig.deployDirectory(HostConfig.java:1185) ~[catalina.jar:9.0.60]
	at org.apache.catalina.startup.HostConfig$DeployDirectory.run(HostConfig.java:1933) ~[catalina.jar:9.0.60]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75) ~[tomcat-util.jar:9.0.60]
	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:118) ~[?:?]
	at org.apache.catalina.startup.HostConfig.deployDirectories(HostConfig.java:1095) ~[catalina.jar:9.0.60]
	at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:477) ~[catalina.jar:9.0.60]
	at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1618) ~[catalina.jar:9.0.60]
	at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:319) ~[catalina.jar:9.0.60]
	at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:123) ~[catalina.jar:9.0.60]
	at org.apache.catalina.util.LifecycleBase.setStateInternal(LifecycleBase.java:423) ~[catalina.jar:9.0.60]
	at org.apache.catalina.util.LifecycleBase.setState(LifecycleBase.java:366) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:946) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:835) ~[catalina.jar:9.0.60]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1396) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1386) ~[catalina.jar:9.0.60]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75) ~[tomcat-util.jar:9.0.60]
	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:140) ~[?:?]
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:919) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:263) ~[catalina.jar:9.0.60]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.StandardService.startInternal(StandardService.java:432) ~[catalina.jar:9.0.60]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:927) ~[catalina.jar:9.0.60]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183) ~[catalina.jar:9.0.60]
	at org.apache.catalina.startup.Catalina.start(Catalina.java:772) ~[catalina.jar:9.0.60]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:345) ~[bootstrap.jar:9.0.60]
	at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:476) ~[bootstrap.jar:9.0.60]
16:11:08.143  ERROR startup.StartupTasksExecutor - FATAL: ERROR: duplicate key value violates unique constraint "tag_inode_pkey"
  Detail: Key (tag_id, inode)=(ab344fa0-c909-4b3c-aa8c-6bacd4f770fa, 7dbb1cb6-1058-46ff-adf4-21bce8f63216) already exists.
com.dotmarketing.exception.DotDataException: ERROR: duplicate key value violates unique constraint "tag_inode_pkey"
  Detail: Key (tag_id, inode)=(ab344fa0-c909-4b3c-aa8c-6bacd4f770fa, 7dbb1cb6-1058-46ff-adf4-21bce8f63216) already exists.
	at com.dotmarketing.startup.AbstractJDBCStartupTask.executeUpgrade(AbstractJDBCStartupTask.java:793) ~[classes/:?]
	at com.dotmarketing.startup.StartupTasksExecutor.executeUpgrades(StartupTasksExecutor.java:199) ~[classes/:?]
	at com.liferay.portal.servlet.MainServlet.init(MainServlet.java:119) ~[classes/:?]
	at org.apache.catalina.core.StandardWrapper.initServlet(StandardWrapper.java:1164) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.StandardWrapper.loadServlet(StandardWrapper.java:1117) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.StandardWrapper.load(StandardWrapper.java:1010) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.StandardContext.loadOnStartup(StandardContext.java:4957) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5264) ~[catalina.jar:9.0.60]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:726) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:698) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:696) ~[catalina.jar:9.0.60]
	at org.apache.catalina.startup.HostConfig.deployDirectory(HostConfig.java:1185) ~[catalina.jar:9.0.60]
	at org.apache.catalina.startup.HostConfig$DeployDirectory.run(HostConfig.java:1933) ~[catalina.jar:9.0.60]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75) ~[tomcat-util.jar:9.0.60]
	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:118) ~[?:?]
	at org.apache.catalina.startup.HostConfig.deployDirectories(HostConfig.java:1095) ~[catalina.jar:9.0.60]
	at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:477) ~[catalina.jar:9.0.60]
	at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1618) ~[catalina.jar:9.0.60]
	at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:319) ~[catalina.jar:9.0.60]
	at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:123) ~[catalina.jar:9.0.60]
	at org.apache.catalina.util.LifecycleBase.setStateInternal(LifecycleBase.java:423) ~[catalina.jar:9.0.60]
	at org.apache.catalina.util.LifecycleBase.setState(LifecycleBase.java:366) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:946) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:835) ~[catalina.jar:9.0.60]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1396) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1386) ~[catalina.jar:9.0.60]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75) ~[tomcat-util.jar:9.0.60]
	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:140) ~[?:?]
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:919) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:263) ~[catalina.jar:9.0.60]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.StandardService.startInternal(StandardService.java:432) ~[catalina.jar:9.0.60]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183) ~[catalina.jar:9.0.60]
	at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:927) ~[catalina.jar:9.0.60]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183) ~[catalina.jar:9.0.60]
	at org.apache.catalina.startup.Catalina.start(Catalina.java:772) ~[catalina.jar:9.0.60]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:345) ~[bootstrap.jar:9.0.60]
	at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:476) ~[bootstrap.jar:9.0.60]
Caused by: org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "tag_inode_pkey"
  Detail: Key (tag_id, inode)=(ab344fa0-c909-4b3c-aa8c-6bacd4f770fa, 7dbb1cb6-1058-46ff-adf4-21bce8f63216) already exists.
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2676) ~[postgresql-42.5.1.jar:42.5.1]
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2366) ~[postgresql-42.5.1.jar:42.5.1]
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:356) ~[postgresql-42.5.1.jar:42.5.1]
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:496) ~[postgresql-42.5.1.jar:42.5.1]
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:413) ~[postgresql-42.5.1.jar:42.5.1]
	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:333) ~[postgresql-42.5.1.jar:42.5.1]
	at org.postgresql.jdbc.PgStatement.executeCachedSql(PgStatement.java:319) ~[postgresql-42.5.1.jar:42.5.1]
	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:295) ~[postgresql-42.5.1.jar:42.5.1]
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:290) ~[postgresql-42.5.1.jar:42.5.1]
	at com.zaxxer.hikari.pool.ProxyStatement.execute(ProxyStatement.java:95) ~[HikariCP-3.4.2.jar:?]
	at com.zaxxer.hikari.pool.HikariProxyStatement.execute(HikariProxyStatement.java) ~[HikariCP-3.4.2.jar:?]
	at com.dotmarketing.common.db.DotConnect.executeStatement(DotConnect.java:264) ~[classes/:?]
	at com.dotmarketing.startup.AbstractJDBCStartupTask.executeUpgrade(AbstractJDBCStartupTask.java:790) ~[classes/:?]
	... 46 more

```


### Steps to Reproduce

Load the custom DB
run
`insert into db_version values (230111, now()) ;`

start dotCMS

### Acceptance Criteria

Database should be updated cleanly.

### dotCMS Version

23.02

### Proposed Objective

Cloud Engineering

### Proposed Priority

Please Select

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Sub-Tasks & Estimates

_No response_

--- a/dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task230119MigrateContentToProperPersonaTagAndRemoveDupTagsTest.java
+++ b/dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task230119MigrateContentToProperPersonaTagAndRemoveDupTagsTest.java
@@ -43,9 +43,16 @@ public class Task230119MigrateContentToProperPersonaTagAndRemoveDupTagsTest {
         APILocator.getTagAPI().addContentletTagInode(fakeHipster,
                 bannerWithTag.getInode(), "tags");
 
+        // let's now create a second content but add the good and bad tag to it. The upgrade task shouldn't fail
+        final Contentlet anotherContentWithTag = TestDataUtils.getBannerLikeContent(true, 1);
+        APILocator.getTagAPI().addContentletTagInode(realHipster,
+                anotherContentWithTag.getInode(), "tags");
+        APILocator.getTagAPI().addContentletTagInode(fakeHipster,
+                anotherContentWithTag.getInode(), "tags");
+
         List<Map<String, Object>> resultsBeforeUT = new DotConnect().setSQL("SELECT tagname\n"
                 + "FROM tag_inode JOIN tag t ON tag_inode.tag_id = t.tag_id\n"
-                + "WHERE inode = '" + bannerWithTag.getInode() + "'").loadObjectResults();
+                + "WHERE inode = '" + bannerWithTag.getInode() + "' ORDER BY tagname DESC").loadObjectResults();
 
         // get(1) since get(0) is a test tag created by the TestDataUtils
         Assert.assertEquals(fakeHipsterName, resultsBeforeUT.get(1).get("tagname"));
@@ -54,7 +61,7 @@ public class Task230119MigrateContentToProperPersonaTagAndRemoveDupTagsTest {
 
         List<Map<String, Object>> resultsAfterUT = new DotConnect().setSQL("SELECT tagname\n"
                 + "FROM tag_inode JOIN tag t ON tag_inode.tag_id = t.tag_id\n"
-                + "WHERE inode = '" + bannerWithTag.getInode() + "'").loadObjectResults();
+                + "WHERE inode = '" + bannerWithTag.getInode() + "' ORDER BY tagname DESC").loadObjectResults();
 
         Assert.assertEquals(realHipsterName, resultsAfterUT.get(1).get("tagname"));
 

--- a/dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task230119MigrateContentToProperPersonaTagAndRemoveDupTags.java
+++ b/dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task230119MigrateContentToProperPersonaTagAndRemoveDupTags.java
@@ -29,7 +29,8 @@ public class Task230119MigrateContentToProperPersonaTagAndRemoveDupTags extends 
             + "               JOIN tag ON tag_inode.tag_id = tag.tag_id\n"
             + "      WHERE tagname LIKE '%:persona') aux,\n"
             + "     tag\n"
-            + "WHERE tag.tagname = REPLACE(aux.tagname, ':persona', '');\n"
+            + "WHERE tag.tagname = REPLACE(aux.tagname, ':persona', '') "
+            + "AND NOT EXISTS (SELECT 1 FROM tag_inode WHERE tag.tag_id = tag_id AND aux.inode = inode);\n"
             + "\n"
             + (DbConnectionFactory.isPostgres() ? DELETE_BAD_RELS_POSTGRES : DELETE_BAD_RELS_MSSQL)
             + "\n"

